SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -595,10 +616,39 @@ void UGlobalStateManager::BecomeAuthoritativeOverAllActors()
 	}
 }
 
+void UGlobalStateManager::BecomeAuthoritativeOverActorsBasedOnLBStrategy()
+{
+	for (TActorIterator<AActor> It(NetDriver->World); It; ++It)
+	{
+		AActor* Actor = *It;
+		if (Actor != nullptr && !Actor->IsPendingKill())
+		{
+			if (Actor->GetIsReplicated() && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor))
+			{
+				Actor->Role = ROLE_Authority;
+				Actor->RemoteRole = ROLE_SimulatedProxy;
+			}
+		}
+	}
+}
+
 void UGlobalStateManager::TriggerBeginPlay()
 {
 	check(GetCanBeginPlay());
 
+	// If we're loading from a snapshot, we shouldn't try and call BeginPlay with authority
+	if (bCanSpawnWithAuthority)
+	{
+		if (GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer)
+		{
+			BecomeAuthoritativeOverActorsBasedOnLBStrategy();
+		}
+		else if (HasAuthority())
```

Comment:
I think you need to check here specifically if this GSM has authority over the STARTUP_ACTOR_MANAGER_COMPONENT_ID.  HasAuthority() checks the SINGLETON_MANAGER_COMPONENT_ID.